printing/printer.mli

```diff
@@ -171,23 +171,22 @@ val pr_transparent_state   : transparent_state -> Pp.t
 
 (** Proofs, these functions obey [Hyps Limit] and [Compact contexts]. *)
 
-val pr_goal                : goal sigma -> Pp.t
+val pr_goal                : ?diffs:(Pp.t option) -> goal sigma -> Pp.t
```

Comment:
fairly strange, why the double option here?

Reply:
Haven't mastered all the details of optional arguments, still doing a lot of trial and error with OCaml.  Doc is sometimes helpful.

Reply:
So if you do:
```ocaml
foo: name:t -> ...
```
`name` will be mandatory and of type `t`

If you do
```ocaml
foo: ?name:t -> ...
```
`name` will be optional and of type `t option`, and set to `None` if omitted. Note that in this case, `foo ~name:x` does set name to `Some x`. So your type above should likely be: `?diffs:Pp.t`.